src/framework/pybind_casts.hpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2021.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_pybind_casts_hpp_
+#define _aer_framework_pybind_casts_hpp_
+
+#include "../simulators/stabilizer/clifford.hpp"
+
+namespace py = pybind11;
+
+namespace pybind11 {
+namespace detail {
+template <typename T> struct type_caster<matrix<T>>{
+  using base = type_caster_base<matrix<T>>;
+public:
+  PYBIND11_TYPE_CASTER(matrix<T>, _("matrix_complex_t"));
+  // Conversion part 1 (Python->C++):
+  bool load(py::handle src, bool convert){
+      // TODO: Check if make sense have to flavors of matrix: F-style and C-style
+      auto py_matrix = py::cast<py::array_t<T>>(src);
```

Comment:
Does this make a copy of src if its already an array?

Reply:
No, as far as I know this just creates the appropriate wrapper to let you manipulate it as an array based object.